ambry-store/src/test/java/com.github.ambry.store/StorageManagerTest.java

```diff
@@ -678,6 +726,40 @@ private void generateConfigs(boolean segmentedLog) {
     storeConfig = new StoreConfig(vProps);
   }
 
+  // unrecognizedDirsOnDiskTest() helpers
+
+  /**
+   * Creates {@code fileCount} files and {@code dirCount} directories at {@code dir}.
+   * @param dir the directory to create the files and dirs at
+   * @param fileCount the number of files to be created
+   * @param dirCount the number of directories to be created
+   * @return the list of files,dirs created as a pair.
+   * @throws IOException
+   */
+  private Pair<List<File>, List<File>> createFilesAndDirsAtPath(File dir, int fileCount, int dirCount)
+      throws IOException {
+    List<File> createdFiles = new ArrayList<>();
+    for (int i = 0; i < fileCount; i++) {
+      File createdFile = new File(dir, "created-file-" + i);
+      if (!createdFile.exists()) {
+        assertTrue("Could not create " + createdFile, createdFile.createNewFile());
+      }
+      createdFile.deleteOnExit();
+      createdFiles.add(createdFile);
+    }
+    List<File> createdDirs = new ArrayList<>();
+    for (int i = 0; i < dirCount; i++) {
+      File createdDir = new File(dir, "created-dir-" + i);
+      assertTrue(createdDir + " exists and is not a directory", !createdDir.exists() || createdDir.isDirectory());
```

Comment:
Correct me if I am wrong. If `createdDir` exists and isDirectory, it can pass this check. And then the `createdDir` might be added into `createdDirs` twice.